src/posthog-surveys.ts

```diff
@@ -99,6 +99,26 @@ export class PostHogSurveys {
                 const targetingFlagCheck = survey.targeting_flag_key
                     ? this.instance.featureFlags.isFeatureEnabled(survey.targeting_flag_key)
                     : true
+
+                let internalTargetingFlagCheck = undefined
+                if (survey.internal_targeting_flag_key) {
+                    // because this feature flag depends on user properties that might not be available yet,
+                    // the decide API will not return this flag when asked for it.
+                    // as a result, we need to distinguish between undefined and false.
+                    // where undefined means that the properties are not yet available.
+                    const value = this.instance.featureFlags.getFeatureFlag(survey.internal_targeting_flag_key)
+                    if (value == undefined) {
+                        internalTargetingFlagCheck = true
+                    } else {
+                        internalTargetingFlagCheck = this.instance.featureFlags.isFeatureEnabled(
```

Comment:
if I'm following this code correctly, this method can return `bool | undefined`, right?  Is the desired behavior that we return false for the flag check if the `isFeatureEnabled` returns undefined?  Idk if this method has the same constraints as the `getFeatureFlag` method from above, but I'm curious why we treat `isFeatureEnabled` as `undefined` if it doesn't resolve to a value when we explicitly say that the other method that could resolve to undefined is `true` if we can't decide the value.

If this is the expected behavior, maybe explain why we're treating it differently here?

Reply:
Yeah, I was confused about this too, and the fix is equally confusing :) . 

The server-side `internal_feature_flag` depends on a person property that is either `survey_responded` or `survey_dismissed`. When we query the feature flags' `decide` endpoint, we do not return a value for this flag if none of those properties are available  yet. And they're only populated if anyone submits a response or dismisses the survey. 

 As a result, an `undefined` value for this flag is not false, its `unknown`, in which case we want to show the survey.

Reply:
Very helpful explanation!  IMO it's worth adding as a comment in here so that future readers of this code block don't have to dig up this PR to understand why we're doing what we're doing :)

Reply:
hmmm, this shouldn't happen 🤔  - the only time we should return undefined here is when the network request to get flags hasn't returned yet, which will happen for all flags.

But, decide will always return all flags with a true/false value, depending on whether they match the condition or not. In this case, if the property doesn't exist, it will return true, since we're doing the 'is not set' operator. Same in the case when the person itself doesn't yet exist.

Given how surveys themselves are loaded via a potentially slower api request, there should really be no need to do any extra finangling here, this should _just work_, and in the worse case, would show up 1.5s later (when we interval recheck for active surveys to show). Did you face any issues while trying this?

Reply:
converting undefined to true can be a bit problematic as it might show the survey to users when we don't intend it to, by default not showing is better than showing many times imo.

Reply:
Yep, this works without having to handle an `undefined` state.